pbft/pbft_failure_detector.cpp

```diff
@@ -15,39 +15,45 @@
 #include <pbft/pbft_failure_detector.hpp>
 #include <utils/bytes_to_debug_string.hpp>
 
-namespace
-{
-    const std::chrono::milliseconds operation_timeout{std::chrono::milliseconds(10000)};
-}
-
 using namespace bzn;
 
-pbft_failure_detector::pbft_failure_detector(std::shared_ptr<bzn::asio::io_context_base> io_context)
+pbft_failure_detector::pbft_failure_detector(std::shared_ptr<bzn::asio::io_context_base> io_context
+    , std::shared_ptr<bzn::options_base> options)
         : io_context(std::move(io_context))
+        , options(options)
         , request_progress_timer(this->io_context->make_unique_steady_timer())
 {
 }
 
 void
 pbft_failure_detector::start_timer()
 {
-    this->request_progress_timer->expires_from_now(operation_timeout);
+    this->request_progress_timer->expires_from_now(this->options->get_fd_oper_timeout());
     this->request_progress_timer->async_wait(std::bind(&pbft_failure_detector::handle_timeout, shared_from_this(), std::placeholders::_1));
 }
 
 void
-pbft_failure_detector::handle_timeout(boost::system::error_code /*ec*/)
+pbft_failure_detector::handle_timeout(boost::system::error_code ec)
 {
+    if (ec)
+    {
+        return;
+    }
+
     std::lock_guard<std::mutex> lock(this->lock);
 
+    assert(!this->ordered_requests.empty());
```

Comment:
This is intentionally a tamer solution (as opposed to the more drastic - stop caring about request ordering at all, only require progress), correct?

Reply:
Actually this is a leftover from some changes I was playing with. The request ordering change didn't show any clear improvements and caused some other complications that made it less desirable.